packages/@aws-cdk/aws-ec2-alpha/lib/route.ts

```diff
@@ -402,6 +444,88 @@ export class NatGateway extends Resource implements IRouteTarget {
   }
 }
 
+/**
+ * Creates a peering connection between two VPCs
+ * @resource AWS::EC2::VPCPeeringConnection
+ */
+export class VPCPeeringConnection extends Resource implements IRouteTarget {
+
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerTargetId: string;
+
+  /**
+   * The VPC peering connection CFN resource.
+   */
+  public readonly resource: CfnVPCPeeringConnection;
+
+  constructor(scope: Construct, id: string, props: VPCPeeringConnectionProps) {
+    super(scope, id);
+
+    const region = props.acceptorRegion ? props.acceptorRegion : Stack.of(this).region;
+    let peerRole: Role | undefined;
+    this.routerType = RouterType.VPC_PEERING_CONNECTION;
+
+    if (props.isCrossAccount && props.acceptorAccountId === undefined) {
+      throw new Error('AcceptorAccountId is required for cross-account peering connections');
+    }
```

Comment:
One thing we try to avoid in CDK is handling dependant properties with error checking. Ideally we want to make it impossible for the user to use a cross account Peering Connection without providing an `acceptorAccountId`.

One thing you could do is turn the `VPCPeeringConnection` into an abstract class (or interface) and have two classes that extend/implement it:

- `CrossAcountVPCPeeringConnection`
- `NonCrossAccountVPCPeeringConnection` (though i don't really like this name)

The `isCrossAccount` property would be safe to remove as that decision is made by the user when deciding which type of Peering Connection the create. The `acceptorAccountId` would be removed from the abstract class so that it can't be set for the `NonCrossAccountVPCPeeringConnection` and added as a required property in `CrossAcountVPCPeeringConnection`.

Reply:
The use case would look like either:
```ts
...
const peeringConnection = new CrossAccountVPCPeeringConnection(this, 'crossAccountCrossRegionPeering', {
  requestorVpc: vpcA,
  acceptorVpc: vpcB,
  acceptorAccountId: '123456789012',
  acceptorRegion: 'us-west-2',
});
```
or
```ts
...
const peeringConnection = new NonCrossAccountVPCPeeringConnection(this, 'nonCrossAccountPeering', {
  requestorVpc: vpcA,
  acceptorVpc: vpcB,
});
```

Reply:
Shikha and I discussed this implementation and we've decided to do an implementation like this:

```
const peeringConnection = new VPCPeeringConnection(this, 'crossAccountCrossRegionPeering', {
  acceptor: new crossAccountAcceptorVPC{
    accountId: '123456789012',
    targetVpc: vpcB,
    targetRegion?: 'us-west-2',
  },
  requestorVpc: vpcA, 
});
```

This way it's intuitive with the L2 construct directly mapping to the L1 construct name, and simple since we need to only instantiate one class, VPCPeeringConnection, which handles both cross-account and same-account scenarios. This method preserves the intuitiveness and simplicity while also eliminating the need to error check.

Reply:
in my opinion the acceptor account id, and region should be concluded from the `acceptorVpc` property, and should not be explicitly provided.